client/modules/IDE/components/AssetList.jsx

```diff
@@ -1,187 +1,63 @@
-import PropTypes from 'prop-types';
-import React from 'react';
-import { connect, useDispatch } from 'react-redux';
-import { bindActionCreators } from 'redux';
-import { Link } from 'react-router-dom';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { Helmet } from 'react-helmet';
-import prettyBytes from 'pretty-bytes';
-import { useTranslation, withTranslation } from 'react-i18next';
-import MenuItem from '../../../components/Dropdown/MenuItem';
-import TableDropdown from '../../../components/Dropdown/TableDropdown';
-
+import { useTranslation } from 'react-i18next';
+import AssetListRow from './AssetListRow';
 import Loader from '../../App/components/loader';
-import { deleteAssetRequest } from '../actions/assets';
 import * as AssetActions from '../actions/assets';
 
-const AssetMenu = ({ item: asset }) => {
+const AssetList = () => {
   const { t } = useTranslation();
-
   const dispatch = useDispatch();
+  const { username, assetList, loading } = useSelector((state) => ({
+    username: state.user.username,
+    assetList: state.assets.list,
+    loading: state.loading
+  }));
```

Comment:
Not a big deal but FYI it's better to have separate `useSelector` calls for each piece of state that you read:

> ### Call useSelector Multiple Times in Function Components
>
> When retrieving data using the useSelector hook, prefer calling useSelector many times and retrieving smaller amounts of data, instead of having a single larger useSelector call that returns multiple results in an object. Unlike mapState, useSelector is not required to return an object, and having selectors read smaller values means it is less likely that a given state change will cause this component to render.
>
> However, try to find an appropriate balance of granularity. If a single component does need all fields in a slice of the state , just write one useSelector that returns that whole slice instead of separate selectors for each individual field.

https://redux.js.org/style-guide/#call-useselector-multiple-times-in-function-components